sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableValuedFunctions.scala

```diff
@@ -103,6 +106,46 @@ object ResolveTableValuedFunctions extends Rule[LogicalPlan] {
           "numPartitions" -> IntegerType) {
         case Seq(start: Long, end: Long, step: Long, numPartitions: Int) =>
           Range(start, end, step, Some(numPartitions))
+      }),
+
+    "show_namespaces" -> Map(
+      /* show_namespaces() */
```

Comment:
This implementation will add a lot of TVFs. How about just add one TVF, something like this:
```sql
SELECT * FROM SELECT_FROM_COMMAND('show_namespaces') WHERE identifier='a' AND pattern = 'b';
```

Reply:
> This implementation will add a lot of TVFs. How about just add one TVF, something like this:
> 
> ```sql
> SELECT * FROM SELECT_FROM_COMMAND('show_namespaces') WHERE identifier='a' AND pattern = 'b';
> ```

In this way its hard to fit where condition to command. and always fetch all data then filter?

If it's show tables, we can't directly filter data in metadata level.